src/gluonnlp/optimizer/lamb.py

```diff
@@ -104,25 +104,34 @@ def update(self, index, weight, grad, state):
             grad = clip(grad, -self.clip_gradient, self.clip_gradient)
 
         mean, var = state
-        mean[:] = self.beta1 * mean + (1. - self.beta1) * grad
-        var[:] = self.beta2 * var + (1. - self.beta2) * square(grad)
+        mean *= self.beta1
+        mean += (1. - self.beta1) * grad
+        var *= self.beta2
+        var += (1. - self.beta2) * square(grad)
 
         r1 = weight.norm()
         if not self.bias_correction:
             r1 = minimum(maximum(r1, self.lower_bound), self.upper_bound)
             g = mean / (sqrt(var) + self.epsilon) + wd * weight
-
         else:
-            # execution bias correction
+            # apply bias correction
             mean_hat = mean / (1. - power(self.beta1, t))
             var_hat = var / (1. - power(self.beta2, t))
-            g = mean_hat / sqrt(var_hat + self.epsilon) + wd * weight
+            var_hat += self.epsilon
+            var_hat[:] = sqrt(var_hat)
+            mean_hat /= var_hat
+            mean_hat += wd * weight
+            g = mean_hat
 
         r2 = g.norm()
 
         # calculate lamb_trust_ratio
-        r = 1. if r1 == 0. or r2 == 0. else r1 / r2
+        ratio = r1 / r2
+        # becomes NaN if ratio == NaN or 0
+        nan_or_zero = ratio / ratio
+        r = where(nan_or_zero, ones_like(ratio), ratio)
```

Comment:
if ratio is not either nan or 0, `where` will return 1, while we want it to return ratio.

Reply:
My bad. The variable name is wrong. Should be nan_or_one

Reply:
Thanks.. I fixed the condition with NaN now: 
```
>>> mx.nd.where(1-nan, mx.nd.array([1],ctx=mx.gpu()), mx.nd.array([0.2],ctx=mx.gpu()))

[1.]
<NDArray 1 @gpu(0)>
>>> mx.nd.where(1-mx.nd.array([1],ctx=mx.gpu()), mx.nd.array([1],ctx=mx.gpu()), mx.nd.array([0.2],ctx=mx.gpu()))

[0.2]
<NDArray 1 @gpu(0)>
```